File: csr/s001/i1.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){
    int n;
    scanf("%d",&n);
    vector<int> a(n);
    for (auto &e:a) scanf("%d",&e);
    int ans=0,l=0,r=0,chk=a[0];
    for(;;) {
        if (chk==n) ans++;
        if (chk<n && r<n-1) {
            r++;
            chk+=a[r];
        } else if (chk>=n) {
            chk-=a[l];
            l++;
        } else {
            break;
        }
    }
    printf("%d\n",ans);
    return 0;
}

File: csr/s001/i1.py
#!/usr/bin/env python3
# -*- coding: UTF-8 -*-

n=int(input())
a=[int(i) for i in input().split()]
chk=a[0]
ans=l=r=0
while 1:
    if chk==n: ans+=1
    if chk<=n and r<n-1:
        r+=1
        chk+=a[r]
    elif chk>n:
        chk-=a[l]
        l+=1
    else:
        break
print(ans)
